### ISSUE TITLE
CIFAR 100 Questions

### ISSUE THREAD

**Issue by HanGuo97** [google/fedjax#262]
Hi, thanks for the awesome library! I want to ask a couple of questions related to CIFAR100 datasets.

1. I noticed that while the dataset is available in the library, the model is not. Curious if a model for CIFAR100 is work-in-progress, or if there is no short-term plan for this?
2. Looking at the CIFAR100 dataset, this seems to be inconsistent with Google's TFF. Notably, the cropping size and normalizing are done differently from TFF. Is this intentional? Would it be correct to say that we could expect this to mirror TFF's design eventually?

Thanks in advance for all the help!

**Comment by jaehunro** [comment]
Hi!

Thanks for reaching out to us about this.

1. The model for CIFAR100 was a work in progress but we decided to put it on hold until there was an explicit request for it. Our initial plan was to add a simple linear model that provides a complete example of all the requirements (e.g. baseline accuracy, hyperparameter configs, etc.) needed to add a model so that other collaborators could follow it as a guideline for contributing. If you're interested, we can work on adding this example PR for a simple model so that you can follow it as a guideline for contributing.
2. We actually are downloading the dataset from the [same source](https://github.com/google/fedjax/blob/main/fedjax/datasets/cifar100.py#L86-L88) as [TFF](https://github.com/tensorflow/federated/blob/b3d8166b51cd3516ae7bd31730f8fcf86e1741c4/tensorflow_federated/python/simulation/datasets/cifar100.py#L101-L103). The only difference is that we apply [additional preprocessing](https://github.com/google/fedjax/blob/main/fedjax/datasets/cifar100.py#L176-L206) on top of the source data following [pytorch-cifar](https://github.com/kuangliu/pytorch-cifar/blob/49b7aa97b0c12fe0d4054e670403a16b6b834ddd/main.py#L30-L40). If you don't want to apply this preprocessing, you could just manually run [fedjax.datasets.cifar100.load_split](https://github.com/google/fedjax/blob/main/fedjax/datasets/cifar100.py#L157-L158).

**Comment by HanGuo97** [comment]
Got it, thanks for letting me know!

**Comment by jaehunro** [comment]
No problem!

I'm also working on some PRs to add an example model for CIFAR-100 and add an option to match the preprocessing done by TFF according to their baseline task [tff.simulation.baselines.cifar100.create_image_classification_task](https://www.tensorflow.org/federated/api_docs/python/tff/simulation/baselines/cifar100/create_image_classification_task)

Hopefully those will help!

**Comment by HanGuo97** [comment]
Oh nice, good to hear -- looking forward to it!

**Comment by jaehunro** [comment]
Just wanted to post an update.
I just merged in #266 that adds the option for matching the preprocessing logic done by TFF.
I also added an example linear model for CIFAR-100 in #265 to show what's needed to check in and contribute a model. 

I've also filed a separate issue #268 to add the ResNet-18 model for CIFAR-100 that's actually used in the ["Adaptive Federated Optimization"](https://arxiv.org/abs/2003.00295) paper. The issue details steps, requirements, etc.
Contributions are very welcomed and encouraged!
